tsne/bh_sne_src/tsne.cpp

```diff
@@ -706,16 +714,60 @@ void run(double* X, int N, int D, double* Y, double perplexity, double theta,
             "%f)!\nLearning embedding...\n",
             (float)(end - start) / CLOCKS_PER_SEC,
             (double)row_P[N] / ((double)N * (double)N));
-  start = clock();
 
-  for (int iter = 0; iter < max_iter; iter++) {
+  TSNEState* state = new TSNEState;
```

Comment:
Certainly here you should be able to use
```suggestion
  unique_ptr<TSNEState> state = make_unique<TSNEState>(
```
with initializer syntax.  Remember there's no types in the C api so the difference between a `unique_ptr` and a `*` pointer doesn't matter there.

Reply:
added a constructor, ported to make_unique.